app/controllers/qa/linked_data_terms_controller.rb

```diff
@@ -7,9 +7,15 @@ class Qa::LinkedDataTermsController < ::ApplicationController
   before_action :check_show_subauthority, :check_id_param, only: :show
   before_action :check_uri_param, only: :fetch
   before_action :validate_auth_reload_token, only: :reload
+  before_action :create_request_header_service, only: [:search, :show, :fetch]
 
   delegate :cors_allow_origin_header, to: Qa::ApplicationController
 
+  class_attribute :request_header_service_class
+  self.request_header_service_class = Qa::LinkedData::RequestHeaderService
```

Comment:
This service is moving some of the parameter processing out of the controller.  It simplifies the controller code and allows the service to focus on the parameter processing and construction of the request_header being passed to search and find.

Reply:
This looks much cleaner 👍